### PR TITLE
Handling Edge Cases for RPC inconsistencies

### PR DESCRIPTION
**Pull request by farhanW3**
## PR-Codex overview
This PR introduces configuration changes related to contract subscriptions, including requery delays. 

### Detailed summary
- Added `contractSubscriptionConfigurationSchema` in `sharedApiSchemas.ts`
- Updated `Config` interface in `config.ts` to include `contractSubscriptionsRequeryDelaySeconds`
- Modified worker tasks in `chainIndexer.ts` and queues in `processEventLogsQueue.ts` and `processTransactionReceiptsQueue.ts` to use `contractSubscriptionsRequeryDelaySeconds`
- Updated cache retrieval in `getConfig.ts` to use `Config` interface
- Updated response schema in routes for contract subscriptions configuration in `get.ts` and `update.ts`

> The following files were skipped due to too many changes: `src/server/routes/configuration/contract-subscriptions/update.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: src/worker/tasks/chainIndexer.ts [hunk 1]
```diff
@@ -42,6 +43,22 @@ export const createChainIndexerTask = async (args: {
...
+          // Sometimes the RPC providers nodes are aware of the latest block
+          // but the block data is not available yet.
+          const block = await provider.getBlockWithTransactions(toBlock);
+
+          if (!block) {
+            logger({
+              service: "worker",
+              level: "warn",
+              message: `Block data not available: ${toBlock} on chain: ${chainId}, url: ${
+                (provider as StaticJsonRpcBatchProvider).connection.url
```
Comment (arcoraven) [inline src/worker/tasks/chainIndexer.ts:56]: We're finding ourselves doing this a few times now. Should prob just have a helper func `getProviderUrl(provider)`

File: src/worker/queues/processTransactionReceiptsQueue.ts [hunk 2]
```diff
@@ -38,9 +38,14 @@ export const enqueueProcessTransactionReceipts = async (
 
   // Enqueue one job immediately and any delayed jobs.
   await _queue.add(jobName, serialized);
-  for (const retryDelay of retryDelays) {
+  for (let i = 0; i < retryDelays.length; i++) {
+    const delay = parseInt(retryDelays[i]) * 1000;
+    const attempts = i === retryDelays.length - 1 ? 10 : 0;
```
Comment (arcoraven) [inline src/worker/queues/processTransactionReceiptsQueue.ts:43]: This is weird enough to warrant a comment:
```
// The last attempt should attempt repeatedly to handle extended RPC issues.
// This backoff attempts at intervals:
// 30s, 1m, 2m, 4m, 8m, 16m, 32m, ~1h, ~2h, ~4h
```

**Review by arcoraven — APPROVED** [review]

**Review by arcoraven — APPROVED** [review]

File: src/server/schemas/sharedApiSchemas.ts [hunk 3]
```diff
@@ -332,3 +332,10 @@ export const walletDetailsSchema = Type.Object({
     Type.Null(),
   ]),
 });
+
+export const contractSubscriptionResponseSchema = Type.Object({
```
Comment (arcoraven) [inline src/server/schemas/sharedApiSchemas.ts:336]: Can you name this `contractSubscriptionConfigurationSchema` to more clearly differentiate it from a Contract Subscription object?

**Review by arcoraven — APPROVED** [review]
Nonblocking feedback

File: src/worker/queues/processEventLogsQueue.ts [hunk 4]
```diff
@@ -33,17 +33,17 @@ export const enqueueProcessEventLogs = async (
   // e.g. 8453:14423125-14423685
   const jobName = `${data.chainId}:${data.fromBlock}-${data.toBlock}`;
   const { contractSubscriptionsRetryDelaySeconds } = await getConfig();
-  const retryDelays = contractSubscriptionsRetryDelaySeconds.split(",");
+  const requeryDelays = contractSubscriptionsRetryDelaySeconds.split(",");
```
[the remark this answers is not in the document]
Comment (farhanW3) [reply src/worker/queues/processEventLogsQueue.ts]: done

**Pull request merged**